bzip2.rs

```diff
@@ -63,6 +63,10 @@ macro_rules! STDOUT {
     };
 }
 
+// work around clippy suggesting using c strings in later versions
```

Comment:
I did rather just suppress the clippy lint using `#[allow]` instead.

Reply:
I believe that causes issues when you do compile with earlier versions where that lint is not defined yet?

Reply:
Regular compilation ignores clippy lints entirely. Only running an older clippy would warn, but I don't see a reason why we should support running an older clippy version.

Reply:
```
warning: unknown lint: `clippy::manual_foo_str_literals`
 --> src/lib.rs:1:10
  |
1 | #![allow(clippy::manual_foo_str_literals)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: did you mean: `clippy::manual_str_repeat`
  |
  = note: `#[warn(unknown_lints)]` on by default
```

so, yeah

Reply:
```
warning: unknown lint: `clippy::manual_c_str_literals`
 --> src/lib.rs:1:10
  |
1 | #![allow(clippy::manual_c_str_literals)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: did you mean: `clippy::manual_str_repeat`
  |
  = note: `#[warn(unknown_lints)]` on by default
```

so yeah

Reply:
right, I've ignored the lint now, so we can use byte string literals, but I do still thing using one constant is nice, the casting stuff is kinda ugly.